[Docker] Dockerfile.dependencies_humble installs older Pydantic version

https://github.com/JdeRobot/RoboticsAcademy/blob/799acfaf260d2f2786f0d1921faf9e4188dee9e4/scripts/mini_RADI/Dockerfile.dependencies_humble#L193
Installs pydantic v1.8.2 which causes issues with RAM

```shell
13:14:31 [Thread-2 (pr] [INFO ] (my_app_logger)  client connected: {'id': 1, 'handler': <websocket_server.websocket_server.WebSocketHandler object at 0x7fce48e79690>, 'address': ('172.17.0.1', 37646)}
13:14:31 [Thread-2 (pr] [INFO ] (my_app_logger)  message received: {"id":"82d6c48b-cf29-405a-a2b2-e9b71b728135","command":"connect"} from client {'id': 1, 'handler': <websocket_server.websocket_server.WebSocketHandler object at 0x7fce48e79690>, 'address': ('172.17.0.1', 37646)}
13:14:33 [MainThread  ] [ERROR] (my_app_logger)  /dev/dri does not exist. No direct GPU access.
13:14:33 [MainThread  ] [INFO ] (my_app_logger)  State changed to connected
13:14:33 [Thread-2 (pr] [INFO ] (my_app_logger)  message received: {"id":"fa417ca1-e69e-4225-a5ae-46f855440344","command":"launch_world","data":{"name":"follow_line_default_ros2","launch_file_path":"/opt/jderobot/Launchers/simple_circuit.launch.py","ros_version":"ROS2","visualization":"gazebo_rae","world":"gazebo","template":"RoboticsAcademy/exercises/static/exercises/follow_line_newmanager/python_template/","exercise_id":"follow_line_newmanager"}} from client {'id': 1, 'handler': <websocket_server.websocket_server.WebSocketHandler object at 0x7fce48e79690>, 'address': ('172.17.0.1', 37646)}
13:14:33 [MainThread  ] [ERROR] (my_app_logger)  'ConfigurationModel' object has no attribute 'model_dump'
Traceback (most recent call last):
  File "//RoboticsAcademy/src/manager/manager/manager.py", line 466, in start
    self.process_messsage(message)
  File "//RoboticsAcademy/src/manager/manager/manager.py", line 367, in process_messsage
    self.trigger(message.command, data=message.data or None)
  File "/usr/local/lib/python3.10/dist-packages/transitions/core.py", line 922, in _get_trigger
    return event.trigger(model, *args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/transitions/core.py", line 402, in trigger
    return self.machine._process(func)
  File "/usr/local/lib/python3.10/dist-packages/transitions/core.py", line 1211, in _process
    return trigger()
  File "/usr/local/lib/python3.10/dist-packages/transitions/core.py", line 416, in _trigger
    self._process(event_data)
  File "/usr/local/lib/python3.10/dist-packages/transitions/core.py", line 439, in _process
    if trans.execute(event_data):
  File "/usr/local/lib/python3.10/dist-packages/transitions/core.py", line 273, in execute
    event_data.machine.callbacks(itertools.chain(event_data.machine.before_state_change, self.before), event_data)
  File "/usr/local/lib/python3.10/dist-packages/transitions/core.py", line 1146, in callbacks
    self.callback(func, event_data)
  File "/usr/local/lib/python3.10/dist-packages/transitions/core.py", line 1165, in callback
    func(event_data)
  File "//RoboticsAcademy/src/manager/manager/manager.py", line 210, in on_launch_world
    self.world_launcher = LauncherWorld(**configuration.model_dump())
AttributeError: 'ConfigurationModel' object has no attribute 'model_dump'

```

--- a/scripts/mini_RADI/Dockerfile.dependencies_humble
+++ b/scripts/mini_RADI/Dockerfile.dependencies_humble
@@ -141,7 +141,7 @@ RUN python3.10 -m pip install --upgrade pip==23.3.1 wheel==0.41.3 setuptools==69
 
 # pip install dependencies
 RUN python3.10 -m pip install \
-    pylint==2.17.4 transitions==0.9.0 pydantic==1.10.8 websocket-client==1.5.2 \
+    pylint==2.17.4 transitions==0.9.0 pydantic==2.4.2 websocket-client==1.5.2 \
     argparse==1.4.0 coverage==6.2 cerberus==1.3.4 empy==3.3.4 jinja2==3.0.3 kconfiglib==14.1.0 \
     matplotlib==3.0.* numpy==1.24.3 nunavut==1.1.0 packaging==21.3 pkgconfig==1.5.5 pyros-genmsg==0.5.8 \
     pyulog==1.0.1 pyyaml==5.4.1 requests==2.31.0 serial==0.0.97 six==1.16.0 toml==0.10.2 psutil==5.9.0 \

